daemon/access.go

```diff
@@ -246,3 +247,32 @@ func (ac interfaceAuthenticatedAccess) CheckAccess(d *Daemon, r *http.Request, u
 
 	return Unauthorized("access denied")
 }
+
+type noticeReadAccess struct{}
+
+func (ac noticeReadAccess) CheckAccess(d *Daemon, r *http.Request, ucred *ucrednet, user *auth.UserState) *apiError {
+	query := r.URL.Query()
+	defaultChecker := openAccess{}
+
+	types, err := sanitizeTypesFilter(query["types"])
+	if err != nil || len(types) == 0 {
+		return defaultChecker.CheckAccess(d, r, ucred, user)
+	}
+
+	for _, noticeType := range types {
+		var checker accessChecker
+		switch noticeType {
+		// TODO: refresh-inhibit and snap-run-inhibit notices must be checked
+		// when they are added.
+		case state.ChangeUpdateNotice:
+			checker = interfaceOpenAccess{Interface: "snap-refresh-inhibit-observe"}
+		default:
+			checker = defaultChecker
```

Comment:
Perhaps instead of using a catch all which ends up being open-access, we should list all types explicitly with a corresponding checker, and in the `default` case is reached deny, i.e. types not accounted for explicitly are denied by default.

Reply:
Updated, thank you!